src/remote-config/remote-config.ts

```diff
@@ -254,6 +293,200 @@ class RemoteConfigTemplateImpl implements RemoteConfigTemplate {
+    for (const key in this.stringifiedDefaultConfig) {
+      configValues[key] = new ValueImpl('default', this.stringifiedDefaultConfig[key]);
+    }
+
+    // Overlays config Value objects derived by evaluating the template.
+    for (const [key, parameter] of Object.entries(this.cache.parameters)) {
+      const { conditionalValues, defaultValue } = parameter;
+
+      // Supports parameters with no conditional values.
+      const normalizedConditionalValues = conditionalValues || {};
+
+      let parameterValueWrapper: RemoteConfigParameterValue | undefined = undefined;
+
+      // Iterates in order over condition list. If there is a value associated
+      // with a condition, this checks if the condition is true.
+      for (const [conditionName, conditionEvaluation] of evaluatedConditions) {
+        if (normalizedConditionalValues[conditionName] && conditionEvaluation) {
+          parameterValueWrapper = normalizedConditionalValues[conditionName];
+          break;
+        }
+      }
+
+      if (parameterValueWrapper && (parameterValueWrapper as InAppDefaultValue).useInAppDefault) {
+        // TODO: add logging once we have a wrapped logger.
+        continue;
+      }
+
+      if (parameterValueWrapper) {
+        const parameterValue = (parameterValueWrapper as ExplicitParameterValue).value;
+        configValues[key] = new ValueImpl('remote', parameterValue);
+        continue;
+      }
+
+      if (!defaultValue) {
+        // TODO: add logging once we have a wrapped logger.
+        continue;
+      }
+
+      if ((defaultValue as InAppDefaultValue).useInAppDefault) {
+        // TODO: add logging once we have a wrapped logger.
+        continue;
+      }
+
+      const parameterDefaultValue = (defaultValue as ExplicitParameterValue).value;
+      configValues[key] = new ValueImpl('remote', parameterDefaultValue);
+    }
+
+    return new ServerConfigImpl(configValues);
+  }
+
+  /**
+   * @returns JSON representation of the server template
+   */
+  public toJSON(): ServerTemplateData {
+    return this.cache;
+  }
+}
+
+class ServerConfigImpl implements ServerConfig {
+  constructor(
+    private readonly configValues: { [key: string]: Value },
+  ){}
+  getBoolean(key: string): boolean {
+    return this.getValue(key).asBoolean();
+  }
+  getNumber(key: string): number {
+    return this.getValue(key).asNumber();
+  }
+  getString(key: string): string {
+    return this.getValue(key).asString();
+  }
+  getValue(key: string): Value {
+    return this.configValues[key] || new ValueImpl('static');
+  }
+}
+
+/**
+ * Remote Config dataplane template data implementation.
+ */
+class ServerTemplateDataImpl implements ServerTemplateData {
+  public parameters: { [key: string]: RemoteConfigParameter };
+  public parameterGroups: { [key: string]: RemoteConfigParameterGroup };
+  public conditions: NamedCondition[];
+  public readonly etag: string;
+  public version?: Version;
+
+  constructor(template: ServerTemplateData) {
```

Comment:
nit: template can be readonly too?

Reply:
That'd make sense, but I'm unsure it's worth another PR. The current approach is comparable to the existing `RemoteConfigTemplateImpl` constructor.

Aside, this reminds me of java's `final`. Most things are readonly, but applying a modifier to each variable ends up being more trouble than it's worth. A single designation, like `const`, is better.